5438: Added override for using locally build docker images

adrian-fitsios opened the pull request.

adrian-fitsios committed:
5438: Added override for using locally build docker images

stuart-bass-cgi commented on local-helm/buyingcatalogue/templates/_helpers.tpl:
```diff
@@ -78,4 +78,15 @@ Create the name of the service account to use
\ No newline at end of file
+{{- end }}
+
+{{/*
```
Can you please put this into each child chart, so that they can stand alone?
In the future, we should look at some sort of child library functions or something

stuart-bass-cgi replied:
Maybe we would be better of specifying a root container registry (which could be local), rather than finding / replacing the dev acr. This would also future proof us a bit, as I guess we are likely to change the dev acr (imho we don't need acr per environment).

adrian-fitsios replied:
> Maybe we would be better of specifying a root container registry (which could be local), rather than finding / replacing the dev acr. This would also future proof us a bit, as I guess we are likely to change the dev acr (imho we don't need acr per environment).

I am afraid that I don't follow what you mean by root container registry.

We could've gone the way where you specify which image you want to pull if you use local images directly in the override, however that'd seem to be more work for little to no benefit as most users don't rename their images. They just build them using the names in their docker-compose files - which means that these names are consistent with the ones in acr, only missing the 'registry tag' (gpitfuturesdevacr.azurecr.io) - hence the replacement approach. 

I do agree that hard-coding the 'gpitfuturesdevacr.azurecr.io' prefix is not the best approach, however it is highly unlikely (at least in my eyes) that this would change.

Would you please elaborate on what you meant by 'root container registry'?

stuart-bass-cgi replied:
gpitfuturesdevacr.azurecr.io - the container registry we're pulling the images from. When we start to apply this out, at the moment, we have a container registry per environment.

adrian-fitsios committed:
5438: PR Comments

Move image.properties tpl from global helpers.tpl to each child chart to ease future splitting of charts
